peers/app_request_network.go

```diff
@@ -42,7 +45,8 @@ type AppRequestNetwork struct {
 func NewNetwork(
 	logLevel logging.Level,
 	registerer prometheus.Registerer,
-	cfg *config.Config,
+	trackedSubnets set.Set[ids.ID],
+	cfg Config,
```

Comment:
Yeah, Config here is an interface instead of a concrete type in order for both Signature Aggregator and the relayer to use it. The value that does get passed in is still technically a pointer that implements the smaller, new network.Config interface